### PR TITLE
CommaApi: use auth.py for request authentication when on PC

### PR DESCRIPTION
**Pull request by iejMac**

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
This should only be done for fetching the routes for replays.

File: selfdrive/ui/qt/api.cc [hunk 1]
```diff
@@ -90,8 +90,15 @@ HttpRequest::HttpRequest(QObject *parent, QString requestURL, const QString &cac
   }
 }
 
-void HttpRequest::sendRequest(QString requestURL){
-  QString token = CommaApi::create_jwt();
+void HttpRequest::sendRequest(QString requestURL, bool create_jwt){
+  if(create_jwt) {
+    token = CommaApi::create_jwt();
+  } else {
+    QString token_json = QString(util::read_file(util::getenv_default("HOME", "/.comma/auth.json", "/.comma/auth.json")).c_str());
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/api.cc:97]: remove c_str()

Comment (iejMac) [reply selfdrive/ui/qt/api.cc:98]: can't initialize QString with std::string

File: selfdrive/ui/replay/replay.hpp [hunk 2]
```diff
@@ -11,6 +11,7 @@
 #include "qt/api.hpp"
 #include "Unlogger.hpp"
 #include "FileReader.hpp"
+#include "common/util.h"
```
Comment (adeebshihadeh) [inline selfdrive/ui/replay/replay.hpp:14]: fix the order

File: selfdrive/ui/replay/replay.cc [hunk 3]
```diff
@@ -3,8 +3,13 @@
 Replay::Replay(QString route_, int seek) : route(route_) {
   unlogger = new Unlogger(&events, &events_lock, &frs, seek);
   current_segment = 0;
+  bool create_jwt = true;
 
-  http = new HttpRequest(this, "https://api.commadotai.com/v1/route/" + route + "/files");
+#ifndef QCOM
```
Comment (adeebshihadeh) [inline selfdrive/ui/replay/replay.cc:8]:
```suggestion
#if !defined(QCOM) && !defined(QCOM2)
```

File: selfdrive/ui/qt/api.hpp [hunk 4]
```diff
@@ -33,14 +33,15 @@ class HttpRequest : public QObject {
   Q_OBJECT
 
 public:
-  explicit HttpRequest(QObject* parent, QString requestURL, const QString &cache_key = "");
+  explicit HttpRequest(QObject* parent, QString requestURL, const QString &cache_key = "", bool create_jwt_=true);
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/api.hpp:36]: whitespace

**Pull request merged**